agent/agent.go

```diff
@@ -37,6 +38,13 @@ type Options struct {
 	HeartbeatInterval  time.Duration
 	AgentSleepInterval time.Duration
 	Cleanup            bool
+	S3Opts             S3Config
+}
+
+type S3Config struct {
```

Comment:
can't we just use the S3Options from pail?

Reply:
also related, perhaps, but why is this public?

Reply:
changed, I had intended to customize this more but it wasn't needed